mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
```

Comment:
> Your statement about tags being global is incorrect for the magical HEAD tags

LOL. My statement also does not apply to the magical unicorns and flying carpets. AFAICT, git HEAD is not a tag. Unlike tags, HEAD belongs to the "heads" namespace:

```
ls .git/refs/
heads  notes  remotes  tags
```

I recommend _not_ using HEAD as the script parameter. To take a snapshot of the latest branch commit, the caller should probably pass the branch name (or that last commit if they need to synchronize across multiple scripts/calls).

This problem remains unresolved.

> git requires archive to be run from inside a clone repository, with a specific branch checkout.

BTW, let's try to limit each Github review comment to one problem. It is easier to keep track of unresolved problems that way.